academic_observatory_workflows/workflows/crossref_fundref_telescope.py

```diff
@@ -16,242 +16,131 @@
         :param catchup: whether to catchup the DAG or not.
-        :param airflow_vars: list of airflow variable keys, for each variable it is checked if it exists in airflow.
-        :param workflow_id: api workflow id.
+        :param gitlab_pool_name: name of the Gitlab Pool.
+        :param gitlab_pool_slots: number of slots for the Gitlab Pool.
+        :param gitlab_pool_description: description for the Gitlab Pool.
         """
 
-        if table_descriptions is None:
-            table_descriptions = {
-                dag_id: "The Funder Registry dataset: " "https://www.crossref.org/services/funder-registry/"
-            }
-
-        if airflow_vars is None:
-            airflow_vars = [
-                AirflowVars.DATA_PATH,
-                AirflowVars.PROJECT_ID,
-                AirflowVars.DATA_LOCATION,
-                AirflowVars.DOWNLOAD_BUCKET,
-                AirflowVars.TRANSFORM_BUCKET,
-            ]
-
-        if load_bigquery_table_kwargs is None:
-            load_bigquery_table_kwargs = {"ignore_unknown_values": True}
-
         super().__init__(
-            dag_id,
-            start_date,
-            schedule_interval,
-            dataset_id,
-            schema_folder,
-            load_bigquery_table_kwargs=load_bigquery_table_kwargs,
-            table_descriptions=table_descriptions,
+            dag_id=dag_id,
+            start_date=start_date,
+            schedule_interval=schedule_interval,
             catchup=catchup,
-            airflow_vars=airflow_vars,
-            workflow_id=workflow_id,
+            airflow_conns=[observatory_api_conn_id],
             tags=[Tag.academic_observatory],
         )
 
+        self.cloud_workspace = cloud_workspace
+        self.bq_dataset_id = bq_dataset_id
+        self.bq_table_name = bq_table_name
+        self.api_dataset_id = api_dataset_id
+        self.schema_folder = schema_folder
+        self.dataset_description = dataset_description
+        self.table_description = table_description
+        self.observatory_api_conn_id = observatory_api_conn_id
+
         # Create Gitlab pool to limit the number of connections to Gitlab, which is very quick to block requests if
         # there are too many at once.
-        pool_name = "gitlab_pool"
-        num_slots = 2
-        description = "A pool to limit the connections to Gitlab."
-        create_pool(pool_name, num_slots, description)
+        create_pool(gitlab_pool_name, gitlab_pool_slots, gitlab_pool_description)
 
         self.add_setup_task(self.check_dependencies)
-        self.add_setup_task(self.get_release_info, pool=pool_name)
-        self.add_task(self.download, pool=pool_name)
+        self.add_setup_task(self.get_release_info, pool=gitlab_pool_name)
+        self.add_task(self.download, pool=gitlab_pool_name)
         self.add_task(self.upload_downloaded)
         self.add_task(self.extract)
```

Comment:
Is the extract task necessary for fundref? The data is pretty small?

Reply:
It could be done in memory. Although extracting it on disk does mean that you can inspect the extracted files if need be.